tests/core/featurizers/test_tracker_featurizer.py

```diff
@@ -49,48 +55,1060 @@ def test_convert_labels_to_ids(domain: Domain):
         assert np.all(expected_array == actual_array)
 
 
+def test_convert_intent_labels_to_ids(domain: Domain):
+    trackers_as_intents = [
+        ['next_intent', 'nlu_fallback', 'out_of_scope', 'restart'],
+        ['greet', 'hello',  'affirm'],
+    ]
+
+    tracker_featurizer = IntentMaxHistoryTrackerFeaturizer()
+
+    actual_labels = tracker_featurizer._convert_labels_to_ids(
+        trackers_as_intents, domain
+    )
+
+    expected_labels = np.array(
+        [
+            [7, 8, 9, 10],
```

Comment:
These numbers depend on `domain.intents.index(...)`. Can we avoid this?

Reply:
Not so important, though.

Reply:
Do you mean it would be better to make `expected_labels` using `domain.intents.index` directly instead of having hard coded it?

Reply:
Yes. Because we don't want to test if the domain has changed, and this test would fail in this case.

Reply:
Ok I made that change in test_convert_action_labels_to_ids and test_convert_intent_labels_to_ids. To be consistent I should probably do that through but I think it could be done as part of the larger tests refactor.